File: Car_rental_spring/src/main/java/ma/ac/emi/services/customer/CustomerService.java
package ma.ac.emi.services.customer;

import ma.ac.emi.dto.BookACarDto;
import ma.ac.emi.dto.CarDto;
import ma.ac.emi.dto.CarDtoListDto;
import ma.ac.emi.dto.SearchCarDto;

import java.util.List;

public interface CustomerService {
    List<CarDto> getAllCars();
    boolean bookACar(BookACarDto bookACarDto);

    CarDto getCarById(long carId);

    List<BookACarDto> getBookingsByUserId(Long userId);

    CarDtoListDto searchCar(SearchCarDto searchCarDto);
}

File: Car_Rental_Front/src/app/modules/customer/services/customer.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../../auth/services/storage/storage.service';

const BASIC_URL = ["http://localhost:8080"];
@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor(private http: HttpClient) { }

  getAllCars(): Observable<any[]> {
      return this.http.get<any>(BASIC_URL + "/api/customer/cars", {
        headers: this.createAuthorizationHeader()
      });
    }

  getCarById(carId : number): Observable<any> {
    return this.http.get<any>(BASIC_URL + "/api/customer/car/"+ carId, {
      headers: this.createAuthorizationHeader()
    });
  }

  bookACar(bookACarDto: any): Observable<any> {
    return this.http.post<any>(BASIC_URL + "/api/customer/car/book", bookACarDto, {
      headers: this.createAuthorizationHeader()
    });
  }

  getBookingsByUserId(): Observable<any> {
    return this.http.get<any>(BASIC_URL + "/api/customer/car/booking/"+ StorageService.getUserId(), {
      headers: this.createAuthorizationHeader()
    });
  }

  searchCar(searchCarDto: any): Observable<any> {
      return this.http.post(BASIC_URL + "/api/customer/car/search", searchCarDto, {
        headers: this.createAuthorizationHeader()
      });
    }
  
  

    createAuthorizationHeader(): HttpHeaders {
        let authHeaders: HttpHeaders = new HttpHeaders();
        return authHeaders.set(
          'Authorization',
          'Bearer ' + StorageService.getToken()
        );
      }
}
